app/models/project.rb

```diff
@@ -929,6 +932,19 @@ def archive!
     update_attribute :status, STATUS_ARCHIVED
   end
 
+  def required_storage
+    Rails.cache.fetch("project##{id}/project_storage", expires_in: 1.hour) do
+      storage, = count_for(:required_project_storage)
```

Comment:
Heh, matter of preference. But I agree your proposal is more explicit, and I like that.